### PR TITLE
Io threads blog

### PR DESCRIPTION
**Pull request by uriyage**
### Description

Addressed PR comments for I/O threads blog post.
Previous PR: https://github.com/valkey-io/valkey-io.github.io/pull/98

### Issues Resolved
-

### Check List
- [ V] Commits are signed per the DCO using `--signoff`

By submitting this pull request, I confirm that my contribution is made under the terms of the BSD-3-Clause License.

### CODE REVIEW

File: content/blog/2024-07-07-aws-to-contribute-efficiency-improvements-for-ValKey-8.md
Comment (madolson) [inline content/blog/2024-07-07-aws-to-contribute-efficiency-improvements-for-ValKey-8.md:43]: Let's include the test setup in text as well, both for accessibility and for search ability.

File: content/blog/2024-07-07-aws-to-contribute-efficiency-improvements-for-ValKey-8.md
Comment (madolson) [inline content/blog/2024-07-07-aws-to-contribute-efficiency-improvements-for-ValKey-8.md:41]: Let's also include the setup here in text so folks can see it without having to read the image description.

File: content/blog/2024-07-07-aws-to-contribute-efficiency-improvements-for-ValKey-8.md
Comment (madolson) [inline content/blog/2024-07-07-aws-to-contribute-efficiency-improvements-for-ValKey-8.md:25]:
```suggestion
This led us down a path of improving the multi-threaded performance of our ElastiCache and MemoryDB services, through a feature we called Enhanced IO [maybe some links to blog posts]. ?
Today we are excited to dive into how we are sharing our learnings from this performance journey by contributing a major performance improvement to the Valkey project.
```

File: content/blog/2024-07-07-aws-to-contribute-efficiency-improvements-for-ValKey-8.md
Comment (madolson) [inline content/blog/2024-07-07-aws-to-contribute-efficiency-improvements-for-ValKey-8.md:48]:
```suggestion
Higher-capacity shards can increase cluster capacity, improve resilience to request surges and provide scaling for hot keys and large collections that can't be addressed by scaling alone. . 
```

I don't know what "simplify key distribution for multi-key transactions" means. So I think we should just merge these two sentence.s

File: content/blog/2024-07-07-aws-to-contribute-efficiency-improvements-for-ValKey-8.md
Comment (madolson) [inline content/blog/2024-07-07-aws-to-contribute-efficiency-improvements-for-ValKey-8.md:44]: I think this paragraph is good, but just in the wrong place. I think we should move it up above `### Major Upgrade to Valkey Performance`.

File: content/blog/2024-07-07-aws-to-contribute-efficiency-improvements-for-ValKey-8.md
Comment (madolson) [inline content/blog/2024-07-07-aws-to-contribute-efficiency-improvements-for-ValKey-8.md:32]: I have another comment suggesting we move `### Benefits of High Capacity Shards` here. If you take that, I think this text should moved into that paragraph.

File: content/blog/2024-07-07-aws-to-contribute-efficiency-improvements-for-ValKey-8.md
Comment (madolson) [inline content/blog/2024-07-07-aws-to-contribute-efficiency-improvements-for-ValKey-8.md:57]:
```suggestion
```

Not sure this sentence adds much.

File: content/blog/2024-07-07-aws-to-contribute-efficiency-improvements-for-ValKey-8.md
Comment (madolson) [inline content/blog/2024-07-07-aws-to-contribute-efficiency-improvements-for-ValKey-8.md:3]: I like this name.

**Review by madolson — COMMENTED** [review]
I think the blog post is a lot closer now, thanks for all your help!

File: content/blog/2024-07-07-aws-to-contribute-efficiency-improvements-for-ValKey-8.md
Comment (avocadosand) [reply content/blog/2024-07-07-aws-to-contribute-efficiency-improvements-for-ValKey-8.md:20]: I do think we need an intro paragraph of some kind. This is a bit too broad and does not tell the reader what to expect in the article or what actions they should expect to take. I would like to see something here like "Read on to learn more about upcoming features as part of Valkey 8.0 release in September 2024." in addition to an overview sentence or two about the project/contributions. Let's drop the subtitle for this section.

File: content/blog/2024-07-07-aws-to-contribute-efficiency-improvements-for-ValKey-8.md
Comment (katiemcollins) [reply content/blog/2024-07-07-aws-to-contribute-efficiency-improvements-for-ValKey-8.md:3]: The title is a bit confusing since Valkey 8.0 isn't being announced with this blog - suggest re-wording.

Comment (katiemcollins) [reply content/blog/2024-07-07-aws-to-contribute-efficiency-improvements-for-ValKey-8.md:3]: Consider updating the title since this is not an announcement blog of Valkey 8.0.

File: content/blog/2024-07-07-aws-to-contribute-efficiency-improvements-for-ValKey-8.md
Comment (katiemcollins) [reply content/blog/2024-07-07-aws-to-contribute-efficiency-improvements-for-ValKey-8.md:20]: minor nit, open source should not have a dash

**Review by avocadosand — COMMENTED** [review]
It's getting there! Main feedback is to identify a call to action, to tell the reader up front what you are going to tell them in detail below, and to summarize what they learned at the bottom. Keep in mind the POV - totally fine that its AWS POV - we want the reader to be able to disambiguate between the two though.

File: content/blog/2024-07-07-aws-to-contribute-efficiency-improvements-for-ValKey-8.md
Comment (madolson) [inline content/blog/2024-07-07-aws-to-contribute-efficiency-improvements-for-ValKey-8.md:19]:
```suggestion
Read on to learn more about the new innovative I/O threading implementation and its impact on performance and efficiency.
```

File: content/blog/2024-07-07-aws-to-contribute-efficiency-improvements-for-ValKey-8.md
Comment (madolson) [inline content/blog/2024-07-07-aws-to-contribute-efficiency-improvements-for-ValKey-8.md:17]:
```suggestion
Valkey 8.0, set for release in September 2024, will bring major performance enhancements through a variety of improvements including a new multi-threaded architecture.
```

File: content/blog/2024-07-07-aws-to-contribute-efficiency-improvements-for-ValKey-8.md
Comment (madolson) [inline content/blog/2024-07-07-aws-to-contribute-efficiency-improvements-for-ValKey-8.md:31]: I still find these two paragraphs really awkward and a lot of what they are saying is already explained elsewhere. The second one is especially a tangent for fork-based snapshot. It doesn't feel like it fits into the narrative well. I know Itay wanted to include reliability in the narrative, I just don't think we are covering it well enough to explain it. Maybe I'll wait for Itay to comment about it.

**Review by madolson — COMMENTED** [review]
Just some minor tweaks on the intro, but other than that looks solid to me. Let's wait for Itay to comment before shipping it out.

File: content/blog/2024-07-07-aws-to-contribute-efficiency-improvements-for-ValKey-8.md
Comment (uriyage) [reply content/blog/2024-07-07-aws-to-contribute-efficiency-improvements-for-ValKey-8.md]: Done

File: content/blog/2024-07-07-aws-to-contribute-efficiency-improvements-for-ValKey-8.md
Comment (touitou-dan) [reply content/blog/2024-07-07-aws-to-contribute-efficiency-improvements-for-ValKey-8.md:31]: In addition to hotkeys, Valkey users with multi-key operations (mget and co) have to ensure that all keys in such operations are located in the same slot. In many cases, the result is that they are using a small number of slots and consequently are restricted on how much their cluster can scale. These users are often looking for larger , more powerful shards

Comment (madolson) [reply content/blog/2024-07-07-aws-to-contribute-efficiency-improvements-for-ValKey-8.md:31]: I guess after reflecting on it some more, I'm inclined to say we should really lean in on the single key/slot use case and say we are accelerating that. We already have a section dedicated to the benefit of bigger shards, so maybe we can move the first paragraph + Dan's added use case there.

Comment (uriyage) [reply content/blog/2024-07-07-aws-to-contribute-efficiency-improvements-for-ValKey-8.md]: I have removed the sentences about performance and efficiency and added Dan's wording to the "Benefits of High Capacity Shards" section.

File: content/blog/2024-07-07-aws-to-contribute-efficiency-improvements-for-ValKey-8.md
Comment (madolson) [inline content/blog/2024-07-07-aws-to-contribute-efficiency-improvements-for-ValKey-8.md:49]:
```suggestion
Throughput increased by approximately 230%, rising from 360K to 1.19M requests per second compared to Valkey 7.2
```

**Review by madolson — APPROVED** [review]
Took a fresh pass, this blog looks really good to me, awesome work! Just waiting on an approval from Itay.

**Review by madolson — APPROVED** [review]

**Pull request merged**
